crates/lsp/src/main_loop.rs

```diff
@@ -378,16 +378,9 @@ impl AuxiliaryState {
         // Set global instance of this channel. This is used for interacting
         // with the auxiliary loop (logging messages or spawning a task) from
         // free functions.
-        unsafe {
-            #[allow(static_mut_refs)]
-            if let Some(val) = AUXILIARY_EVENT_TX.get_mut() {
-                // Reset channel if already set. Happens e.g. on reconnection after a refresh.
-                *val = auxiliary_event_tx;
-            } else {
-                // Set channel for the first time
-                AUXILIARY_EVENT_TX.set(auxiliary_event_tx).unwrap();
-            }
-        }
+        AUXILIARY_EVENT_TX
+            .set(auxiliary_event_tx)
+            .expect("Auxiliary event channel can't be set more than once.");
```

Comment:
Notably you only see the `expect()` error when running with `-- --nocapture`.

This is because if the auxiliary or main loop threads panic, then we don't propagate the panic up to the main process at all. Tokio _will_ return a `JoinError` if we actually await the `JoinSet` of the aux and main loops, we just need to do so in `start_server()` while we also do `server.serve(service).await;`. I couldn't easily figure that out here so I'm saving that for another PR.

But note that with this PR `cargo test -- --nocapture` no longer shows any panics, but did reliably before.